tools/runner/properties.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2022 gRPC authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runner
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// LogInfo contains infomation for a log entry.
+type LogInfo struct {
+	podNameElement string
+	containerName  string
+	logPath        string
+}
+
+// NewLogInfo creates a pointer of new LogInfo object.
+func NewLogInfo(podNameElement string, containerName string, logPath string) *LogInfo {
+	return &LogInfo{
+		podNameElement: podNameElement,
+		containerName:  containerName,
+		logPath:        logPath,
+	}
+}
+
+// PodLogProperties creates container log property name to
+// container log link map.
+func PodLogProperties(logInfos []*LogInfo, logURLPrefix string) map[string]string {
+	properties := make(map[string]string)
+	for _, logInfo := range logInfos {
+		prefix := []string{logInfo.podNameElement, "name", "log", logInfo.containerName}
```

Comment:
I don´t think it needs "name" here.

I might add ...prefixes as a list of prefixes at the end of the arguments.

The "prefix" being caluclated in the function is not a prefix, it is the key.

I would just have key := strings.Join(append(prefix, logInfo.podNameElement, "log", logInfo.containerName), ".")

In fact, I might make this into a separate function, PodLogPropertyKey() with arguments of type LogInfo and ...prefix.

Then in the runner you can pass the prefix "pod" as an additional argument to put the properties in the right place.